spec/migrations/20210519204247_move_miq_set_memberships_spec.rb

```diff
@@ -0,0 +1,77 @@
+require_migration
+
+describe MoveMiqSetMemberships do
+  let(:rel_stub)        { migration_stub(:Relationship) }
+  let(:membership_stub) { migration_stub(:MiqSetMembership) }
+  let(:ns_stub)         { Class.new(ActiveRecord::Base) { self.table_name = "miq_ae_namespaces" } }
+  let(:rp_stub)         { Class.new(ActiveRecord::Base) { self.table_name = "resource_pools" } }
+
+  migration_context :up do
+    context "two sets with one member each" do
+      it 'converts relationship records to set membership table records' do
+        ns1 = ns_stub.create!
+        ns2 = ns_stub.create!
+        ns1_child = ns_stub.create!
+        ns2_child = ns_stub.create!
+        ns1_rel = create_rel(ns1, nil)
+        ns2_rel = create_rel(ns2, nil)
+        create_rel(ns1_child, ancestry_for(ns1_rel))
+        create_rel(ns2_child, ancestry_for(ns2_rel))
+        create_resource_pool
+
+        migrate
+
+        expect(membership_stub.find_by(:member_id => ns1_child.id)).to have_attributes(:miq_set_id => ns1.id, :member_type => "MiqAeNamespace")
+        expect(membership_stub.find_by(:member_id => ns2_child.id)).to have_attributes(:miq_set_id => ns2.id, :member_type => "MiqAeNamespace")
+
+        expect(rel_stub.count).to eq(2) # remaining ResourcePool relationships
+      end
+    end
+  end
+
+  migration_context :down do
+    context "two sets with one member each" do
+      it 'converts set membership table records to relationship records' do
+        ns1 = ns_stub.create!
+        ns2 = ns_stub.create!
+        ns1_child = ns_stub.create!
+        ns2_child = ns_stub.create!
+        membership_stub.create!(:miq_set_id => ns1.id, :member_type => "MiqAeNamespace", :member_id => ns1_child.id)
+        membership_stub.create!(:miq_set_id => ns2.id, :member_type => "MiqAeNamespace", :member_id => ns2_child.id)
+        create_resource_pool
+
+        migrate
+
+        # Ensure we now are back to 4 records.  2 parents and 2 children.
+        parent_rel1 = rel_stub.find_by(:resource_id => ns1.id)
+        parent_rel2 = rel_stub.find_by(:resource_id => ns2.id)
+        child_rel1  = rel_stub.find_by(:resource_id => ns1_child.id)
+        child_rel2  = rel_stub.find_by(:resource_id => ns2_child.id)
+        expect(parent_rel1).to have_attributes(:relationship => "membership", :resource_type => "MiqAeNamespace", :ancestry => nil)
+        expect(parent_rel2).to have_attributes(:relationship => "membership", :resource_type => "MiqAeNamespace", :ancestry => nil)
```

Comment:
I don't think this is right?  The root node's resource_type should be the set type, not the type of the child records.  It's confusing in this particular spec because MiqAeNamespace is not an existing set type.  If we change this to, say, ScanItemSet and ScanItem, I'd expect the root nodes to be ScanItemSet and the child nodes to be ScanItem.

Reply:
That is probably true.  I was going off what was here, but I am completely fine changing the underlying models to something that makes a bit more sense (and is easier to grok.  Will change.